### PR TITLE
TT-1607 Allow email in verification code

### PR DESCRIPTION
**Pull request by depsiatwal**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.
 - [x] (if changing the UI) Add a printscreen to the PR

### CODE REVIEW

File: enrolment/forms.py [hunk 1]
```diff
@@ -132,6 +132,13 @@ class UserAccountVerification(forms.Form):
         error_messages={'required': MESSAGE_INVALID_CODE}
     )
 
+    def __init__(self, *args, **kwargs):
+        super(UserAccountVerification, self).__init__(*args, **kwargs)
+        if self.initial.get('email') is None:
+            self.fields['email'] = fields.EmailField(
+                label='Your email address'
+            )
```
Comment (richtier) [inline enrolment/forms.py]: can you also please add a comment next to the `email = ` at line 127 to include that the field can be overridden in `__init__`

File: enrolment/forms.py [hunk 2]
```diff
@@ -132,6 +132,13 @@ class UserAccountVerification(forms.Form):
         error_messages={'required': MESSAGE_INVALID_CODE}
     )
 
+    def __init__(self, *args, **kwargs):
+        super(UserAccountVerification, self).__init__(*args, **kwargs)
```
Comment (richtier) [inline enrolment/forms.py]: `super().__init__(*args, **kwargs)`

File: enrolment/views.py [hunk 3]
```diff
@@ -146,7 +146,10 @@ def get_context_data(self, *args, **kwargs):
 class RestartOnStepSkipped:
     def render(self, *args, **kwargs):
         prev = self.steps.prev
-        if prev and not self.get_cleaned_data_for_step(prev):
+        current = self.steps.current
+        if prev and not self.get_cleaned_data_for_step(prev) and not (
+                current == VERIFICATION
+        ):
```
Comment (richtier) [inline enrolment/views.py]: this mixin is aimed to be general use, but this is making `RestartOnStepSkipped` very specific to the "user verification code" case.

please move this logic to the user verification code view :)

**Comment by depsiatwal** [comment]
image

File: enrolment/tests/test_forms.py [hunk 4]
```diff
@@ -98,6 +100,20 @@ def test_create_user(mock_create_user):
     assert form.cleaned_data["user_details"] == data
 
 
+def test_verification_code_empty_email():
+
+    form = forms.UserAccountVerification()
+    assert type(form.fields['email']) is fields.EmailField
```
Comment (richtier) [inline enrolment/tests/test_forms.py]: `assert isinstance(form.fields['email'], fields.EmailField)` [why](https://stackoverflow.com/a/1549854/904887)

**Pull request merged**